code/game/antagonist/antagonist_update.dm

```diff
@@ -15,6 +15,10 @@
 		spawn(3)
 			var/mob/living/carbon/human/H = player.current
 			if(istype(H)) H.change_appearance(APPEARANCE_ALL, H.loc, H, valid_species, state = GLOB.z_state)
+	else if(!preserve_appearance && (flags & ANTAG_SET_APPEARANCE_H))
+		spawn(3)
```

Comment:
`spawn(3)` does exactly the same in both, can be shortened to one `spawn` before or after